optimism/src/main.rs

```diff
@@ -77,6 +85,10 @@ pub fn main() -> ExitCode {
 
     let mut env = mips_witness::Env::<ark_bn254::Fr>::create(cannon::PAGE_SIZE as usize, state, po);
 
+    // Install signal catcher
+    let term = Arc::new(AtomicBool::new(false));
+    signal_hook::flag::register(signal_hook::consts::SIGTERM, Arc::clone(&term)).unwrap();
```

Comment:
This seems exactly equivalent to the code on lines 68-70, I assume that's not on purpose?